### PR TITLE
[MSUE-142]

### PR DESCRIPTION
**Pull request by binish-exalture**
Rewrite tests in .NET lib to avoid using tokens/keys that are real or look like real(https://sift.atlassian.net/browse/MSUE-142)

There were no such hard coded tokens or keys in the .NET Sift client project. 

I have moved the request object JSON(s) to single folder(with multiple JSON for each request object) and completed the test project successfully by consuming the files from a single place. The current test project was targeting the .NET Core version 2.0, which is out of support. So I changed the target version of the same to .NET Core 2.1 assuming that there can be still customers using this version as it was an LTS version of .NET Core.

I have added these many Test Projects for targeting different versions of .Net Core, .NET, and .NET Framework:
1) Sift.Test.Core3.1 for targeting .NET Core 3.1
2) Sift.Test.Core2.1 for targeting .NET Core 2.1(Existing Test Project renamed from "Test" to make it meaningful)
3) Sift.Test.Net6 for targeting .NET 6
4) Sift.Test.Framework4.8 for targeting .NET Framework 4.8

Now all the above tests projects are consuming the same requests JSON(s) I mentioned above.

### CODE REVIEW

**Comment by Toji-Thomas** [comment]
@viaskal-sift PR looks good to merge.

**Comment by binish-exalture** [comment]
Updated cake script to include all the four test projects. Also added xunit.runner.visualstudio packages for the cake test for .net framework 4.8 test project.

Before the cake script was taking dlls for packaging from the Debug configuration, now changed to Release

File: build.cake [hunk 1]
```diff
@@ -30,10 +51,64 @@ Task("generate")
...
+    var dotNetFramework4_8TestProject = "./Sift.Test.Framework4.8/Sift.Test.Framework4.8.csproj";
+
+    MSBuild(dotNetFramework4_8TestProject, new MSBuildSettings
+    {
+        Configuration = configuration,
+        PlatformTarget = PlatformTarget.MSIL,
+        Verbosity = Verbosity.Minimal,
+        MSBuildPlatform = MSBuildPlatform.Automatic,
+        ToolVersion = MSBuildToolVersion.VS2017,
+        ToolPath = EnvironmentVariable("MSBUILD_TOOL_PATH") ?? "C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/MSBuild/Current/Bin/MSBuild.exe"
```
Comment (viaskal-sift) [inline build.cake:82]: @binish-exalture looks like some hardcoded thing, is it generic enough?

File: build.cake [hunk 2]
```diff
@@ -30,10 +51,64 @@ Task("generate")
...
+    .DoesForEach(testProjects, (project) =>
+{
+    var testAdapterPath = System.IO.Path.Combine(System.IO.Path.GetDirectoryName(solutionFile), "packages");
+    if (project.Contains("Sift.Test.Framework4.8.csproj"))
+    {
+        var testAssembly = System.IO.Path.Combine(System.IO.Path.GetDirectoryName(project), "bin", configuration, "Sift.Test.Framework4.8.dll");
+
+        VSTest(testAssembly, new VSTestSettings
+        {
+            ToolPath = EnvironmentVariable("VSTEST_TOOL_PATH") ?? "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\Common7\\IDE\\CommonExtensions\\Microsoft\\TestWindow\\vstest.console.exe",
```
Comment (viaskal-sift) [inline build.cake:97]: the same here as above

**Pull request closed without merging**